catalog/dags/common/slack.py

```diff
@@ -421,13 +421,13 @@ def on_failure_callback(context: dict) -> None:
 @task
 def notify_slack(
     text: str,
-    dag_id: str,
     username: str = "Airflow Notification",
     icon_emoji: str = ":airflow:",
+    dag: DAG | None = None,
 ) -> None:
     send_message(
         text,
         username=username,
         icon_emoji=icon_emoji,
-        dag_id=dag_id,
+        dag_id=dag.id,
```

Comment:
I think this is supposed to be [`dag_id`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/dag/index.html#airflow.models.dag.DAG.dag_id) and not `id`.

```suggestion
        dag_id=dag.dag_id,
```

This suggests to me that we should have some tests for this because it slipped through our CI + CD workflow.